src/snowflake/snowpark/dataframe.py

```diff
@@ -2258,6 +2259,101 @@ def sort(
 
             return df
 
+    @df_api_usage
+    @publicapi
+    def sort_by_all(
```

Comment:
I'm wondering if we really need a new API to do order by all. Maybe we can use the existing API. For instance, `df.sort()` without a column can mean order by all. cc @sfc-gh-aling